src/pages/recentDeliveries/RecentDeliveriesPage.tsx

```diff
@@ -0,0 +1,71 @@
+import "./RecentDeliveries.css";
+import Container from "react-bootstrap/Container";
+import RecentDeliveriesTable from "./RecentDeliveriesTable";
+import { Request, ExampleQueryDocument } from "../../generated/graphql";
+import { useQuery } from "@apollo/client";
+
+export class RequestWithDate {
+  smileRequest: Request;
+  totalSamples: number;
+  firstDelivered: string;
+  lastDateUpdated: string;
+  constructor(smileRequest: Request) {
+    this.smileRequest = smileRequest;
+    this.totalSamples = smileRequest.hasSampleSamples.length;
+    let deliveryDates = this.resolveAndSortDeliveryDates(smileRequest);
+    this.firstDelivered = deliveryDates[0];
+    this.lastDateUpdated = deliveryDates[1];
+  }
+  private resolveAndSortDeliveryDates(request: Request) {
+    var smDataList: string[] = [];
+    for (var i = 0; i < request.hasSampleSamples.length; i++) {
```

Comment:
basically we want to never use native looping.  use map, filter, reduce or lodash funcions